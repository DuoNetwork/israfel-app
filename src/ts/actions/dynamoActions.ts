import * as CST from 'ts/common/constants';
import { IStatus, IUserOrder, VoidThunkAction } from 'ts/common/types';
import util from 'ts/common/util';
import dynamoUtil from '../../../../israfel-relayer/src/utils/dynamoUtil';

export function statusUpdate(status: IStatus[]) {
	return {
		type: CST.AC_STATUS,
		value: status
	};
}

export function scanStatus(): VoidThunkAction {
	return async dispatch => {
		const states = await dynamoUtil.scanStatus();
		dispatch(statusUpdate(states));
	};
}

export function userOrdersUpdte(userOrders: IUserOrder[]) {
	return {
		type: CST.AC_USER_ORDERS,
		value: userOrders
	};
}

export function getUserOrders(): VoidThunkAction {
	return async (dispatch, getState) =>
		dispatch(
			userOrdersUpdte(
				await dynamoUtil.getUserOrders(
					getState().web3.account,
					util.getUTCNowTimestamp() - 30 * 86400000
				)
			)
		);
}

export function refresh(): VoidThunkAction {
	return dispatch => {
		dispatch(scanStatus())
		dispatch(getUserOrders());
	}
}
