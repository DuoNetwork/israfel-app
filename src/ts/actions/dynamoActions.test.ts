import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import dynamoUtil from '../../../../israfel-relayer/src/utils/dynamoUtil';
import * as dynamoActions from './dynamoActions';

const mockStore = configureMockStore([thunk]);

describe('actions', () => {
	test('statusUpdate', () => {
		expect(dynamoActions.statusUpdate([{ test: 'test' }] as any)).toMatchSnapshot();
	});

	test('scanStatus', () => {
		const store: any = mockStore({});
		dynamoUtil.scanStatus = jest.fn(() =>
			Promise.resolve({
				test: 'test'
			})
		);
		store.dispatch(dynamoActions.scanStatus());
		return new Promise(resolve =>
			setTimeout(() => {
				expect(store.getActions()).toMatchSnapshot();
				resolve();
			}, 0)
		);
	});

	test('userOrdersUpdte', () => {
		expect(dynamoActions.userOrdersUpdte([{ test: 'test' }] as any)).toMatchSnapshot();
	});

	test('getUserOrders', () => {
		const store: any = mockStore({
			web3: {
				account: '0x0'
			}
		});
		dynamoUtil.getUserOrders = jest.fn(() =>
			Promise.resolve([{
				test: 'test'
			}])
		);
		store.dispatch(dynamoActions.getUserOrders());
		return new Promise(resolve =>
			setTimeout(() => {
				expect(store.getActions()).toMatchSnapshot();
				resolve();
			}, 0)
		);
	});
});
