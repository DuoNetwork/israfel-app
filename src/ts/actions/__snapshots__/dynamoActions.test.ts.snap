// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions getUserOrders 1`] = `
Array [
  Object {
    "type": "userOrders",
    "value": Array [
      Object {
        "test": "test",
      },
    ],
  },
]
`;

exports[`actions scanStatus 1`] = `
Array [
  Object {
    "type": "status",
    "value": Object {
      "test": "test",
    },
  },
]
`;

exports[`actions statusUpdate 1`] = `
Object {
  "type": "status",
  "value": Array [
    Object {
      "test": "test",
    },
  ],
}
`;

exports[`actions userOrdersUpdte 1`] = `
Object {
  "type": "userOrders",
  "value": Array [
    Object {
      "test": "test",
    },
  ],
}
`;
